src/Container.js

```diff
@@ -177,49 +190,54 @@ export default class Container {
             instance,
             resolver;
         context = this._resolverContext.beginResolve(name);
+
         try {
-            if (registration.dependencyList !== undefined) {
-                for (let i = 0, len = registration.dependencyList.length; i < len; i++) {
-                    dependencyKey = registration.dependencyList[i];
-                    if (utils.isString(dependencyKey)) {
-                        if(this.isGroupRegistered(dependencyKey)) {
-                            dependency = this.resolveGroup(dependencyKey);
+            if (typeof registration.externallyOwnedFactory === "function") {
+                instance = registration.externallyOwnedFactory.call(this);
```

Comment:
don't think it makes sense re bind the `this` context as we invoke the factory. We should pass 'this' as the container would be an argument, but best not to screw with a factoy we were given